### ISSUE TITLE
Npc's wont "finish" construction in outpost

### ISSUE THREAD

**Issue by jasonmbrown** [CleverRaven/Cataclysm-DDA#35581]
# Describe the bug
Npcs seem to get "bored" of constructing, they will do some then stop.

# Steps To Reproduce

1) Create an outpost.
2) Create a Large construction zone (I used "cut grass).
3) Set an npc to do construction tasks
4) Wait a while, and npc will get bored before finishing all the work

Additionally if you have an npc doing construction and also do some yourself, they suddenly get motivated to finish your work, and try to shove you out of the way.

# Expected behavior

Npc's should finish building in the zones provided. If they can't for some reason, they should say why (Missing tool or too tired or something. If you talk with them)

# Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-9422-g8d5b04e [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Modular Turrets [modular_turrets],
    Salvaged Robots [Salvaged_Robots],
    Alternative Map Key [alt_map_key],
    Mutant NPCs [mutant_npcs],
    More City Locations [cityside],
    More Locations [more_locations],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Roadheader and other mining vehicles [Heavy miners],
    Tanks and Other Vehicles [Tanks],
    Prevent Zombie Revivication [no_reviving_zombies],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Simplified Nutrition [novitamins],
    No Fungal Monsters [No_Fungi]
]

**Comment by Funguss** [comment]
Hmm. I've not been able to get them to cut grass at all, but yes, they also occasionally stop working because: 
1. If there are overlapping (or possibly multiple) zones.
2. If there are items on the construction.
3. If they're out of sight of the player.

I've not been able to reliably reproduce one and three so I've been holding off on creating a bug report. Number 1. seems to happen randomly. I have a feeling 3. happens when it's dark and they don't have a light source but the location is lit, or the area to deconstruct is outside the range of their light source so they think it's too dark to deconstruct. It's difficult to reliably reproduce, though. Number 2. can actually be useful, but for the most part it's just annoying, especially when deconstructing beds.

**Comment by davidpwbrown** [comment]
If you can pull the linked PR locally, and test to see if this fixes your issue?

**Comment by Funguss** [comment]
Erp... Sadly, I have no idea what I'm doing when it comes to compiling. I've tried and failed many times. Even my little "Hello world!" programs fail to compile. I once wrote a calculator in C that I got to compile and run, but it treated every "y" as "n" and so refused to tell you what it had actually calculated. I made it frustratingly cheerful, too, and got it to say it was made by me. The thing was a total embarrassment.

Short answer, no, I can't. :'|
